pkg/server/tls_windows_test.go

```diff
@@ -0,0 +1,295 @@
+package server
+
+import (
+	"crypto"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"github.com/github/smimesign/certstore"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"regexp"
+	"testing"
+	"time"
+)
+
+func TestEasyFilter(t *testing.T) {
+	c := &winCertStoreHandler{
+		cfg: WindowsCertificateFilter{
+			ServerStore:       "My",
+			ServerSystemStore: "LocalMachine",
+		},
+	}
+	serverSt := newFakeStore()
+	sc := makeCert(time.Now().Add(time.Duration(-1)*time.Minute), time.Now().Add(5*time.Minute), []int{1, 2, 3}, "", "")
+	serverSt.identities = append(serverSt.identities, newFakeIdentity(sc))
+	findCert := func(systemStore, _ string) (certstore.Store, error) {
+		return serverSt, nil
+	}
+	serverIdentity, err := c.findCertificate(c.cfg.ServerSystemStore, c.cfg.ServerStore, c.cfg.ServerIssuerCommonNames, c.cfg.ServerTemplateID, nil, findCert)
+	require.NoError(t, err)
+	foundCert, err := serverIdentity.Certificate()
+	require.NoError(t, err)
+	require.NotNil(t, serverIdentity)
```

Comment:
Should this be moved directly below line 30? Ditto for other instances of this check in the file